## Import wooldridge library

In [1]:
import wooldridge as woo

## Load dataset

In [2]:
df = woo.data("jtrain2")
df.head()

train  age  educ  black  hisp  married  nodegree  mosinex  re74  re75  \
0      1   37    11      1     0        1         1       13   0.0   0.0   
1      1   22     9      0     1        0         1       13   0.0   0.0   
2      1   30    12      1     0        0         0       13   0.0   0.0   
3      1   27    11      1     0        0         1       13   0.0   0.0   
4      1   33     8      1     0        0         1       13   0.0   0.0   

       re78  unem74  unem75  unem78  lre74  lre75     lre78  agesq  mostrn  
0   9.93005       1       1       0    0.0    0.0  2.295566   1369      13  
1   3.59589       1       1       0    0.0    0.0  1.279792    484      13  
2  24.90950       1       1       0    0.0    0.0  3.215249    900      13  
3   7.50615       1       1       0    0.0    0.0  2.015723    729      13  
4   0.28979       1       1       0    0.0    0.0 -1.238599   1089      13

## (i) use an indicator variable $train$ to determine the proportion of men who received vocational training

In [3]:
# Sum of train (number of men who received training)
# Divided by total number of observations
proportion_train = df['train'].sum() / len(df)

print(f"Proportion of men who received job training: {proportion_train:.4f}")

Proportion of men who received job training: 0.4157


## (ii) The variable $re78$ represents 1978 earnings, measured in thousands of 1982 dollars. Find the means of $re78$ for a sample of men who received vocational training and for those who did not. Is the difference economically large?

In [4]:
# Mean earnings in 1978 for trained and not trained men
mean_re78 = df.groupby('train')['re78'].mean()

# Calculate the difference
diff = mean_re78[1] - mean_re78[0]

print("Average earnings in 1978 (in thousands of 1982 dollars):")
print(mean_re78)
print(f"\nDifference (trained - not trained): {diff:.2f}")

Average earnings in 1978 (in thousands of 1982 dollars):
train
0    4.554802
1    6.349145
Name: re78, dtype: float64

Difference (trained - not trained): 1.79


Men who received training in 1978 earned an average of $1,790 more (in 1982 dollars) than those who did not.

This difference represents almost 40% of the average wage of those without training (1.79/4.55 ≈ 0.39).

Conclusion: The difference is economically significant

## (iii) The variable $unem78$ is an indicator of whether a man was unemployed or not in 1978. What proportion of men who received vocational training are unemployed? And among those who did not receive training? Comment on the difference.


In [5]:
# Calculate unemployment proportions by training status
unem_proportions = df.groupby('train')['unem78'].mean()

print("Proportion unemployed in 1978:")
print(unem_proportions)

# Difference between trained and not trained
diff_unem = unem_proportions[1] - unem_proportions[0]
print(f"\nDifference (trained - not trained): {diff_unem:.4f}")

Proportion unemployed in 1978:
train
0    0.353846
1    0.243243
Name: unem78, dtype: float64

Difference (trained - not trained): -0.1106


The difference of 11 percentage points is economically relevant, indicating that those who were trained had a lower risk of unemployment.

## (iv) From items (ii) and (iii), does the professional training program appear to have been effective? What would make your conclusions more convincing?

(ii) Average earnings in 1978:
- Trained: US 6,350
- Untrained: US 4,550
- Difference: + US 1,790 → ~40% increase

(iii) Unemployment rate in 1978:
- Trained: 24.3%
- Untrained: 35.4%
- Difference: -11 percentage points

The figures strongly suggest that vocational training may have increased wages and reduced unemployment.

But without correcting for selection bias and without checking statistical significance, we cannot say that the effect is causal.